docs/typescript.md

```diff
@@ -149,18 +179,18 @@ console.log(youngestUser); // { name: "...", age: maxAge }
    When iterating over enum keys using `Object.keys`, the resulting array is of type `string[]`. However, if you want to access enum member values directly, you might need a type assertion.
 
 ```typescript
-enum Direction {
-  Up,
-  Down,
-  Left,
-  Right,
-}
-
-const directions: Direction[] = [Direction.Up, Direction.Down];
-
-for (const key of directions) {
-  const direction = directions[key];
-  console.log(direction); // Output: Direction.Up, Direction.Down
+import { getKnownPropertyNames } from '@metamask/utils';
+
+enum Direction { 
+    Up = 'up', 
+    Down = 'down', 
+    Left = 'left', 
+    Right = 'right', 
+} 
+const directions: Direction[] = [Direction.Up, Direction.Down]; 
+for (const key of getKnonwPropertyNames(direction)) {
+    const direction = directions[key as keyof typeof directions];
+    console.log(direction); // Output: Direction.Up, Direction.Down }
```

Comment:
I should have clarified that my `getKnownPropertyNames` example wasn't meant to be used here. I'm sorry for the confusion.

The usefulness of `getKnownPropertyNames` is that it makes the `as` assertion *unnecessary* ([you can see why by checking out its definition](https://github.com/MetaMask/utils/blob/ec0237cf6793b9b31efffde247152c12ffd1a260/src/misc.ts#L102-L117)). So the above code can't be used as an example for this section.

We'll need to remove the `getKnownPropertyNames` import statement. The type annotation for `directions` can also be removed (this is preferable for reasons detailed [here](https://github.com/MetaMask/contributor-docs/pull/74/files#diff-ae03a9f2eb638a25c1d8d7ba853c78220f1ed44a19ad97a8d780cbc0c00a4260R7-R46)).

```ts
enum Direction { 
  Up = 'up', 
  Down = 'down', 
  Left = 'left', 
  Right = 'right', 
}
const directions = [Direction.Up, Direction.Down];
```

From there, we can resolve the index signature error that arises from the property access expression `directions[key]`,
> Element implicitly has an 'any' type because index expression is not of type 'number'.(7015)

by choosing one of these two options:

- option 1:
```ts
for (const key of Object.keys(direction)) {
  const direction = directions[key as keyof typeof directions];
}
```
- option 2:
```ts
for (const key of Object.keys(directions) as (keyof typeof directions)[]) {
  const direction = directions[key];
}
```

You can use either or both as the example(s) for this entry.

Reply:
Thank you @MajorLift for the comprehensive explanation, modified this section as well